metric_writer.py

```diff
@@ -41,36 +42,46 @@ def process_redis_server(redis_server, redis_port, redis_namespace):
     if not lines:
         return 0
 
-    for tries in range(5):
-        params = { "p" : "root", "db": "service-metrics", "u": "root" }
-        r = requests.post("http://%s:%d/write" % (config.INFLUX_SERVER, config.INFLUX_PORT), params=params, data=lines)
-        if r.status_code in (200, 204):
-            return count
-
-        if str(r.status_code)[0] == '4':
-            log.error("Cannot write metric due to 4xx error. %s" % r.text)
-            return 0
-
-        log.warning("Cannot write metric due to other error Retyring. %s" % r.text)
-        sleep(3)
-    else:
-        log.error("Data points not submitted due to repeated submission problems.")
-
-    return 0
+    params = {"p": "root", "db": "service-metrics", "u": "root"}
+    timeout_notification = monotonic() + 3600
+    while True:
+        try:
+            r = requests.post("http://%s:%d/write" % (config.INFLUX_SERVER,
+                                                      config.INFLUX_PORT), params=params, data=lines)
+            if r.status_code in (200, 204):
+                return count
+            elif str(r.status_code)[0] == '4':
+                log.error("Cannot write metric due to 4xx error. %s" % r.text)
+                return 0
+            else:
+                error = r.text
+        except Exception as error:
```

Comment:
Should we report this error, rather than ignore it?

Reply:
Right, this'll get reported here https://github.com/metabrainz/metric-writer/pull/1/files#diff-ee05b30870a75fa527c6edf230064a4c7b13deb7588de32080bd9451d662ca08R67